From 5179d84bfb8ce3751bdf6f1c76935141cc591db6 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:28:01 -0400
Subject: [tools] perf/top: Add missing newline if the 'uid' is invalid

Message-id: <1408462094-14194-313-git-send-email-jolsa@redhat.com>
Patchwork-id: 88302
O-Subject: [PATCH RHEL7.1 BZ1131394 312/325] perf top: Add missing newline if the 'uid' is invalid
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit ea432a8bb940e6bea2aaeca3c0ff3d931ad81f2e
Author: Ingo Molnar <mingo@kernel.org>
Date: Wed Nov 13 00:26:09 2013 +0100

description
===========
Add missing newline if the 'uid' is invalid:

  hubble:~> perf top --stdio -u help
  Error:
  Invalid User: helphubble:~>

Fixed by this patch:

  comet:~/tip/tools/perf> perf top --stdio -u help
  Error:
  Invalid User: help
  comet:~/tip/tools/perf>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-top.c b/tools/perf/builtin-top.c
index b8f8e29..71e6402 100644
--- a/tools/perf/builtin-top.c
+++ b/tools/perf/builtin-top.c
@@ -1172,7 +1172,7 @@ int cmd_top(int argc, const char **argv, const char *prefix __maybe_unused)
  status = target__validate(target);
  if (status) {
   target__strerror(target, status, errbuf, BUFSIZ);
-  ui__warning("%s", errbuf);
+  ui__warning("%s\n", errbuf);
  }
 
  status = target__parse_uid(target);
@@ -1180,7 +1180,7 @@ int cmd_top(int argc, const char **argv, const char *prefix __maybe_unused)
   int saved_errno = errno;
 
   target__strerror(target, status, errbuf, BUFSIZ);
-  ui__error("%s", errbuf);
+  ui__error("%s\n", errbuf);
 
   status = -saved_errno;
   goto out_delete_evlist;
-- 
1.7.1